adr/adr-responses-api.md

```diff
@@ -0,0 +1,45 @@
+
+## Context
+
+Red Hat is developing an agentic capability for its AI platforms, including RHEL AI and Red Hat OpenShift AI (RHOAI). The core reasoning engine will leverage the Llama Stack. We are faced with a choice between two available APIs within Llama Stack for implementing agentic logic:
+
+1. **Llama Stack Agents API:** An older API that provides an explicit `Agent` construct on the server. While this simplifies the management of agent definitions for the client, this API is not aligned with the broader AI ecosystem's direction.
+2. **Llama Stack Responses API:** A newer API designed for compatibility with the OpenAI Responses API format. This API accepts a list of tools and a chat history and returns a model-generated response, which may include one or more tool calls. It does not provide a server-side `Agent` construct, requiring the client to manage the agent abstraction (tools, workflows, etc.).
+
+The strategic direction from the Llama Stack team is to prioritize the Responses API to maximize compatibility with the vast ecosystem of tools, SDKs, and developer knowledge built around the OpenAI API de facto standard. The older Agents API could be deprecated, heavily refactored, or receive minimal future investment, posing a significant long-term risk.
+
+The core architectural question is: How can we build a robust, developer-friendly agentic framework on an API that does not provide an explicit `Agent` construct without sacrificing the usability that such a construct provides?
+
+One issue that is *not* in scope for this document and is not affected by this decision is whether tools will be invoked by the server or by the client.  For both APIs, it is the case that:
+
+* MCP tools are invoked by the server.  The Responses API can be configured to get approval from the client before invoking the tool, but with either configuration, the tool is invoked by the server.
+* "Hosted" or "built-in" tools are also invoked by the server.
+* "Function" tools are invoked by the client.
+
+So security and network access concerns for tool calling are outside of the scope of this document because they are the same concerns regardless of which API is used.
+
+## Decision
+
+We will adopt the OpenAI-compatible **Llama Stack Responses API** as the exclusive server interface for agentic reasoning within Red Hat's AI products.
+
+To address the absence of a server-side agent construct, we will need **a client-side Agent construct**. The decision about its form is addressed in [ADR: Adopt a Minimal SDK and Ecosystem Strategy for Agents](https://github.com/opendatahub-io/agents/pull/2). Some proposals include adding the functionality to the official Llama Stack client library, creating a new SDK maintained by Red Hat (perhaps as part of OpenDataHub), providing a container image with a simple HTTP API, and adopting a pre-existing SDK such as LangChain. That ADR will cover the pros and cons of these options and propose a final resolution.
```

Comment:
I would say 'may want' rather than 'will need'.

Reply:
Hmm.  I believe all of the proposals listed in #2 currently include a client-side agent construct.  Do you have an alternative proposal for that one?  Or is the point just to leave it open here and resolve it there?

Reply:
I'm saying, that in my view, the absence of a server side agent construct does not mean we then *need* a client-side Agent. We may or may not want one, which is the subject of #2. (In my view saying 'will need' is implicitly making a decision on #2).

Reply:
I agree with @grs. Decisions about client side agents are outside of the scope of this ADR. I think with that correction, I am ready to approve this.

Reply:
@jwm4 once you push these changes up, I am ready to approve this